k8s/client_test.go

```diff
@@ -31,116 +29,208 @@ func (s *K8sTestSuite) SetupSuite() {
 			}
 		},
 	}
+
+	k, _ := client.CreateDeployment(context.Background(), "test_image", cloud.CreatedCoreInstance{}, &apiv1.ServiceAccount{}, true, true)
```

Comment:
This is doesn't look part of the setup. And the deployment variable is only used by `TestUpdateDeploymentByLabel` so I would create the deployment only there.

Reply:
I wanted TestUpdateDeploymentByLabel to only test that function. So I'd leave it like it is.